File: app/src/main/java/me/pwcong/jpstart/rxbus/event/EventContainer.java
package me.pwcong.jpstart.rxbus.event;

/**
 * Created by Pwcong on 2016/10/3.
 */

public class EventContainer {

    public static final int TYPE_PHOTOVIEW = 123;
    public static final int TYPE_SETTING = 456;
    public static final int TYPE_GAME = 666;


    private int type;
    private Object event;

    public EventContainer(int type, Object event) {
        this.type = type;
        this.event = event;
    }

    @Override
    public String toString() {
        return "EventContainer{" +
                "type=" + type +
                ", event=" + event +
                '}';
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public Object getEvent() {
        return event;
    }

    public void setEvent(Object event) {
        this.event = event;
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/rxbus/event/EventContainer.kt
package com.github.pwcong.jpstart.rxbus.event

class EventContainer(var type: Int, var event: Any) {
    override fun toString(): String {
        return "EventContainer{" +
                "type=" + type +
                ", event=" + event +
                '}'
    }

    companion object {
        const val TYPE_PHOTOVIEW: Int = 123
        const val TYPE_SETTING: Int = 456
        const val TYPE_GAME: Int = 666
    }
}
